src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -194,61 +194,65 @@ protected override void BeginProcessing()
         protected override void ProcessRecord()
         {
             bool result = true;
-            string resolvedPath = string.Empty;
             string jsonToParse = string.Empty;
 
             if (Json != null)
             {
                 jsonToParse = Json;
             }
 
-            if (Path != null)
+            else if (Path != null)
             {
-                resolvedPath = PathUtils.ResolveFilePath(Path, this, _isLiteralPath);
+                string resolvedPath = PathUtils.ResolveFilePath(Path, this, _isLiteralPath);
 
-                if (!System.IO.File.Exists(resolvedPath))
+                if (!File.Exists(resolvedPath))
                 {
                     ItemNotFoundException exception = new(
-                        resolvedPath,
+                        Path,
                         "PathNotFound",
                         SessionStateStrings.PathNotFound);
 
-                    ThrowTerminatingError(new ErrorRecord(
-                        exception.ErrorRecord,
-                        exception));
+                    ThrowTerminatingError(exception.ErrorRecord);
                 }
 
                 jsonToParse = File.ReadAllText(resolvedPath);
             }
 
-            try
+            if (string.IsNullOrWhiteSpace(jsonToParse))
             {
-                var parsedJson = JToken.Parse(jsonToParse);
-
-                if (_jschema != null)
+                result = false;
```

Comment:
Just have `return false` for this `if` block, so you can remove the else block to reduce nesting.

Reply:
@daxian-dbw Hmm I think I can't return boolean since this is inside `void BeginProcessing()`. I could `WriteObject` then `return` if you prefer?